docs/reference/mapping/params/subobjects.asciidoc

```diff
@@ -109,26 +109,137 @@ PUT my-index-000001/_doc/metric_1
 <1> The entire mapping is configured to not support objects.
 <2> The document does not support objects
 
+Setting `subobjects: false` disallows the definition of <<object,`object`>> and <<object,`nested`>> sub-fields, which
+can be too restrictive in cases where it's desirable to have <<nested,`nested`>> objects or sub-objects with specific
+behavior (e.g. with `enabled:false`). In this case, it's possible to set `subobjects: auto`, so subobjects are allowed
+to be pre-defined in the mapping (including in dynamic templates), while dynamically-defined fields do not introduce
+intermediate objects. For instance:
```

Comment:
Just a suggestion here but I think we can simplify and not mention differences between statically defined mappings and dynamic mappings, as there isn't really a difference.

```suggestion
behavior (e.g. with `enabled:false`). In this case, it's possible to set `subobjects: auto`, which auto-flattens <insert anchor to the auto-flattening paragraph> whenever possible and falls back to creating an object mapper otherwise (instead of rejecting the mapping as `subobjects: false` does). For instance:
```